config/initializers/sidekiq.rb

```diff
@@ -3,13 +3,22 @@
 require 'three_scale/sidekiq_retry_support'
 require 'three_scale/sidekiq_logging_middleware'
 
+REDIS_PARAMS_WHITELIST = %i[username password db id timeout read_timeout write_timeout connect_timeout ssl
+                            custom ssl_params driver protocol client_implementation command_builder inherit_socket
+                            reconnect_attempts middlewares circuit_breaker sentinels sentinel_password
+                            sentinel_username role name url].freeze
+
+def sanitize_redis_config(cfg)
+  cfg.slice(*REDIS_PARAMS_WHITELIST)
+end
+
 Sidekiq::Client.try(:reliable_push!) unless Rails.env.test?
 
 Rails.application.config.to_prepare do
   Sidekiq.configure_server do |config|
     config.try(:reliable!)
 
-    config.redis = ThreeScale::RedisConfig.new(System::Application.config.sidekiq).config
+    config.redis = ThreeScale::RedisConfig.new(sanitize_redis_config(System::Application.config.sidekiq)).config
```

Comment:
```suggestion
    config.redis = sanitize_redis_config(ThreeScale::RedisConfig.new(System::Application.config.sidekiq).config)
```

Reply:
Sorry, I actually have no idea if this will work, or correct, but my thinking is that we want to filter out stuff just before passing it to Sidekiq. Because maybe `ThreeScale::RedisConfig` can add something to the config, after it was previously sanitized (because we might be interested in some extra details in for our own redis connections).

Reply:
Also, I don't know if we could actually ditch `config.sidekiq`... it seems to be the same as `System::Application.config.redis`, and to me `config.redis` just makes more sense, because the config as such doesn't have anything sidekiq-specific...

Reply:
> Because maybe ThreeScale::RedisConfig can add something to the config, after it was previously sanitized

That's a good point. If I apply the suggestion, then `config.redis=` will receive a `Hash` rather than an `ActiveSupport::OrderedOptions` like before. I tried and everything seems to work fine. It should be safe since `config.redis=` is supposed to receive a `Hash`, according to [docs](https://github.com/sidekiq/sidekiq/wiki/Using-Redis#using-an-initializer).

I implemented this in https://github.com/3scale/porta/pull/3810/commits/4ab0f860aa4fb81d4e2dcd19da2dd39b16448db3.

> Also, I don't know if we could actually ditch config.sidekiq

It seems superfluous. I removed it in https://github.com/3scale/porta/pull/3810/commits/8caf7758c94545d225e32d8a0a5e4090de7cac67.